docs/src/main/asciidoc/optaplanner.adoc

```diff
@@ -0,0 +1,1034 @@
+            Lesson attachedLesson = Lesson.findById(lesson.getId());
+            attachedLesson.setTimeslot(lesson.getTimeslot());
+            attachedLesson.setRoom(lesson.getRoom());
+        }
+    }
+
+}
+----
++
+For simplicity's sake, this code handles only one `TimeTable`,
+but it is straightforward to enable multi-tenancy and handle multiple `TimeTable` instances of different high schools in parallel.
++
+The `getTimeTable()` method returns the latest time table from the database.
+It uses the `ScoreManager` (which is automatically injected)
+to calculate the score of that time table, so the UI can show the score.
++
+The `solve()` method starts a job to solve the current time table and store the time slot and room assignments in the database.
+It uses the `SolverManager.solveAndListen()` method to listen to intermediate best solutions
+and update the database accordingly.
+This enables the UI to show progress while the backend is still solving.
+
+. Adjust the `TimeTableResourceTest` accordingly, now that the `solve()` method returns immediately.
+Poll for the latest solution until the solver finishes solving:
++
+[source,java]
+----
+package org.acme.optaplanner.rest;
+
+import javax.inject.Inject;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.acme.optaplanner.domain.Lesson;
+import org.acme.optaplanner.domain.TimeTable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.optaplanner.core.api.solver.SolverStatus;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@QuarkusTest
+public class TimeTableResourceTest {
+
+    @Inject
+    private TimeTableResource timeTableResource;
+
+    @Test
+    @Timeout(600_000)
+    public void solveDemoDataUntilFeasible() throws InterruptedException {
+        timeTableResource.solve();
+        TimeTable timeTable = timeTableResource.getTimeTable();
+        while (timeTable.getSolverStatus() != SolverStatus.NOT_SOLVING) {
+            // Quick polling (not a Test Thread Sleep anti-pattern)
+            // Test is still fast on fast machines and doesn't randomly fail on slow machines.
+            Thread.sleep(20L);
+            timeTable = timeTableResource.getTimeTable();
+        }
+        assertFalse(timeTable.getLessonList().isEmpty());
+        for (Lesson lesson : timeTable.getLessonList()) {
+            assertNotNull(lesson.getTimeslot());
+            assertNotNull(lesson.getRoom());
+        }
+        assertTrue(timeTable.getScore().isFeasible());
+    }
+
+}
+----
+
+. Build an attractive web UI on top of these REST methods to visualize the timetable.
+
+Take a look at the `optaplanner-quickstart` source code
+in https://github.com/quarkusio/quarkus-quickstarts[the Quarkus quickstarts repository]
+to see how this all turns out.
```

Comment:
This should be at the beginning as mentioned above.

Reply:
I am adding an anchor link to back to the Solution section here, so readers do find the sources if need be.